### DISCUSSION TITLE
What does the metrics 'leader.election' represent?

### DISCUSSION THREAD

**Discussion by J-Swift** [sidekiq/sidekiq#6503] [Q&A] [answered]
I had assumed it was the count of elections run, but it seems like it might be the total runtime to complete the election process? I can't find any reference to it in my searching.

**Comment by mperham** [comment]
It's fired by a Sidekiq process when it gains leadership of the cluster. Leadership normally should only change after a deploy so if you see this outside of a deploy window, it's usually a symptom of the previous leader process having a serious problem (e.g. crashing)

**Reply by J-Swift** [reply]
To be clear, Im wondering what the value of the “leader.election” _metric_ represents for the statsd integration.

**Comment by mperham** [comment, marked as answer]
It’s a mark in time. You can use it just like a deployment marker to draw a vertical line in a graph to track election frequency, if that’s something you’d like to display.

**Reply by J-Swift** [reply]
~Hmm, ok that makes more sense on why the values were so 'off'. Thats a bit surprising though, wouldnt just reporting the metric _at a certain time_ with a value of 1 give you a more well-behaved metric in terms of rollup etc? Just thinking out loud, thanks for clarifying.~

EDIT: it looks like that _is_ what this is. my datadog had some weird filtering applied so it was displaying values that were extremely abnormal. Thanks again!
